### PR TITLE
P2P reconnect logic

### PR DESCRIPTION
**Pull request by rorp**

### CODE REVIEW

File: node/src/main/scala/org/bitcoins/node/networking/P2PClient.scala [hunk 1]
```diff
@@ -195,6 +208,24 @@ case class P2PClientActor(
...
 
+  private def reconnect() = {
+    currentPeerMsgHandlerRecv = initPeerMsgHandlerReceiver
+
+    reconnectionTry = reconnectionTry + 1
+    if (reconnectionTry >= maxReconnectionTries) {
+      logger.error("Exceeded maximum tries to reconnect")
+      context.stop(self)
+    } else {
+      val delay = reconnectionDelay * reconnectionTry
```
Comment (benthecarman) [inline node/src/main/scala/org/bitcoins/node/networking/P2PClient.scala]: we probably want this exponential, not linear increasing

**Comment by Christewart** [comment]
I tried this out with the `MaxConnectionsTest` test case I wrote and it doesn't seem to pass: https://github.com/Christewart/bitcoin-s-core/tree/2021-08-19-p2p_reconnect

**Comment by Christewart** [comment]
Running this this morning gives me logs that repeatedly output `Attempting to process duplicate filter` 

>2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:29:21UTC WARN [ChainHandler] Attempt to process duplicate filters
2021-08-19T12:30:15UTC INFO [DataMessageHandler] Received maximum amount of filters in one batch. This means we are not synced, requesting more
2021-08-19T12:30:15UTC INFO [PeerMessageSender] Requesting compact filters from FilterSyncMarker(startHeight=193000, stopBlockHash=000000000000075cb11ae6bd4762ecf63585b7925effe38f3849ee9798f3affd)

**Comment by Christewart** [comment]
I did a couple hours worth of IBD on this this morning and had no issues. I believe this solves #3397

**Pull request merged**